src/ORM/FieldType/DBForeignKey.php

```diff
@@ -63,11 +72,17 @@ public function scaffoldFormField($title = null, $params = null)
         $list = DataList::create($hasOneClass);
         // Don't scaffold a dropdown for large tables, as making the list concrete
         // might exceed the available PHP memory in creating too many DataObject instances
-        if ($list->count() < 100) {
+        $threshold = self::config()->get('dropdownFieldThreshold');
+        if ($list->count() < $threshold) {
             $field = new DropdownField($this->name, $title, $list->map('ID', $titleField));
             $field->setEmptyString(' ');
         } else {
             $field = new NumericField($this->name, $title);
+            $field->setRightTitle(
```

Comment:
I don’t think we should show a message like this to content authors

Reply:
How about admins? Part of the complaint was visibility over the issue. Happy for it to be wrapped in `if ($isAdmin)`?

Reply:
Or maybe just something more generic - remove the bit about the config var maybe. It'll save a fair bit of time for both parties if there is an "error message"